Added Doctor and Booking models with relationships

HassanSheraz opened the pull request:
- Created Doctor and Booking tables in SQLAlchemy
- Defined relationships between User, Doctor, and Booking models
- Added API endpoints for creating & fetching doctors and bookings
- Updated database initialization to include new tables

HassanSheraz committed:
Added Doctor and Booking models with relationships

HassanSheraz commented:
@aduobed  Could you review this PR? It's ready for approval. Thanks!

aduobed commented:
I am not seeing the changes to create new tables for doctor and booking. There should be a new sql file to create the tables.

aduobed commented:
Also your changes do not have any new endpoints for bookings and doctors. There should be a new file in the router's directory to hold the api. Do you want to work on that or you want me to work on that?

HassanSheraz committed:
Added Doctor and Booking table creation in init.sql

HassanSheraz commented:
I have added and updated `init.sql`, including the Doctor and Booking tables.
Successfully tested in MySQL, this PR is now complete and ready for review.

aduobed commented on mysql-init/init.sql:
```diff
@@ -14,4 +15,23 @@ CREATE TABLE IF NOT EXISTS users (
+);
+
+-- Create Doctors table
+CREATE TABLE IF NOT EXISTS doctors (
```
Can we have this in a different sql file and not the init.sql file? This is a best practice to have new tables creation different so we don't touch the old file.

HassanSheraz committed:
added script script to create db and user, removed old file and creating seperate ones

HassanSheraz commented:
@aduobed 

- I have started splitting `init.sql` into separate files as requested.
- Added `01_create_database.sql` for database & user creation.
- Committing the rest step by step.

HassanSheraz committed:
Added script to create Users table

HassanSheraz commented:
- I have added `02_create_users_table.sql` to define the Users table separately.
- This is a separate commit from the database creation.
- Continuing with Doctors and Bookings next.

HassanSheraz committed:
Added seperate script to create Doctors table

HassanSheraz commented:
- I have added `03_create_doctors_table.sql` to define the Doctors table separately.
- This is a separate commit from Users and Database creation.
- Continuing with Bookings next.

HassanSheraz committed:
Added script to create Bookings table

HassanSheraz commented:
- I have added `04_bookings.sql` to define the Bookings table separately.
- Now, all tables (Database, Users, Doctors, Bookings) are in separate files as requested.

Please review the changes.
